test/pool.test.ts

```diff
@@ -45,30 +80,103 @@ abstract class PoolUser {
 
   async setReceiver(receiver: this, weight: number): Promise<void> {
     const receivers = await this.getAllReceivers();
+
     await submit(this.pool.setReceiver(receiver.addr, weight), "setReceiver");
-    if (weight == 0) {
-      receivers.delete(receiver.addr);
-    } else {
-      receivers.set(receiver.addr, weight);
-    }
-    await this.expectReceivers(receivers);
-  }
 
-  async getAllReceivers(): Promise<Map<string, number>> {
-    const receivers = await this.pool.getAllReceivers();
-    return new Map(receivers.map(({receiver, weight}) => [receiver, weight]));
+    updateReceiversList(
+      receivers,
+      receiver,
+      (receiverWeight) => (receiverWeight.receiverWeight = weight)
+    );
+    await this.expectReceivers(receivers);
   }
 
   async expectSetReceiverReverts(
     receiver: this,
     weight: number,
     expectedCause: string
   ): Promise<void> {
+    const receivers = await this.getAllReceivers();
     await submitFailing(
       this.pool.setReceiver(receiver.addr, weight),
       "setReceiver",
       expectedCause
     );
+    await this.expectReceivers(receivers);
+  }
+
+  async setProxy(receiver: this, weight: number): Promise<void> {
+    const receivers = await this.getAllReceivers();
+    await submit(this.pool.setProxy(receiver.addr, weight), "setProxy");
+    updateReceiversList(
+      receivers,
+      receiver,
+      (receiverWeight) => (receiverWeight.proxyWeight = weight)
+    );
+    await this.expectReceivers(receivers);
+  }
+
+  async expectSetProxyReverts(
+    receiver: this,
+    weight: number,
+    expectedCause: string
+  ): Promise<void> {
+    const receivers = await this.getAllReceivers();
+    await submitFailing(
+      this.pool.setProxy(receiver.addr, weight),
+      "setProxy",
+      expectedCause
+    );
+    await this.expectReceivers(receivers);
```

Comment:
I don’t think this check is necessary. This is guaranteed by the EVM and slows down the test.

Reply:
Good catch :D Fixed